src/main/java/seedu/parser/Parser.java

```diff
@@ -0,0 +1,222 @@
+    public static final Pattern ADD_COMMAND_FORMAT = Pattern.compile(
+            "n/(?<itemName>.+)" + "\\s+" +
+                    "sn/(?<serialNumber>.+)" + "\\s+" +
+                    "t/(?<equipmentType>.+)" + "\\s+" +
+                    "c/(?<cost>.+)" + "\\s+" +
+                    "pf/(?<purchasedFrom>.+)" + "\\s+" +
+                    "pd/(?<purchasedDate>.+)"
+    );
+    public static final Pattern VIEW_COMMAND_FORMAT = Pattern.compile("n/(?<itemName>.+)");
+    public static final Pattern DELETE_COMMAND_FORMAT = Pattern.compile("s/(?<itemName>.+)");
+    // Pattern extracts first n-1 tags
+    public static final Pattern ARGUMENT_FORMAT = Pattern.compile(
+            "((?:sn|n|t|c|pf|pd)\\/[\\w\\s\\-]+?)\\s+(?=sn|n|t|c|pf|pd)"
+    );
+    // Extracts last tag
+    public static final Pattern ARGUMENT_TRAILING_FORMAT = Pattern.compile(
+            "(?<!\\w)(?:sn|n|t|c|pf|pd)\\/([\\w\\s\\-]+)"
+    );
+    public static final String MESSAGE_INCOMPLETE_COMMAND_MISSING_DELIMITER =
+            "Please split your command into arguments with each argument seperated by spaces!";
+
+    public Command parseCommand(String userInput) throws IncompleteCommandException {
+        ArrayList<String> commandAndArgument = null;
+        ArrayList<String> args = null;
+        try {
+            commandAndArgument = splitCommandTerm(userInput);
+        } catch (IncompleteCommandException e) {
+            return new IncorrectCommand(MESSAGE_INCOMPLETE_COMMAND_MISSING_DELIMITER);
+        }
+
+        switch (commandAndArgument.get(0)) {
+        case AddCommand.COMMAND_WORD:
+            try {
+                args = prepareAdd(commandAndArgument.get(1));
+                return new AddCommand(args);
+            } catch (IncompleteCommandException e) {
+                return new IncorrectCommand(AddCommand.COMMAND_WORD + AddCommand.COMMAND_DESCRIPTION);
+            }
+            break;
+        case CheckCommand.COMMAND_WORD:
+            try {
+                args = prepareView(commandAndArgument.get(1));
+                return new CheckCommand(args);
+            } catch (IncompleteCommandException e) {
+                return new IncorrectCommand(CheckCommand.COMMAND_WORD + CheckCommand.COMMAND_DESCRIPTION);
+            }
+            break;
+        case DeleteCommand.COMMAND_WORD:
+            try {
+                args = prepareDelete(commandAndArgument.get(1));
+                return new DeleteCommand(args);
+            } catch (IncompleteCommandException e) {
+                return new IncorrectCommand(DeleteCommand.COMMAND_WORD + DeleteCommand.COMMAND_DESCRIPTION);
+            }
+            break;
+        case UpdateCommand.COMMAND_WORD:
+            try {
+                args = extractArguments(commandAndArgument.get(1));
+                UpdateCommand updateCommand = new UpdateCommand(serialNumber);
```

Comment:
the serialNumber here is not defined previously

Reply:
Should I pass in a serialNumber into the constructor? Since any update will require that anyway.

Reply:
Yes sure that'll work! But just have to define it before passing it into the constructor :D

Reply:
My bad haha, missed out quite a lot stuff trying to push this one out